### PR TITLE
Callbacks can now handle string with a class and a method Class#Method

### PR DESCRIPTION
**Pull request by cybey**
#3

### CODE REVIEW

**Comment by managr** [comment]
That's a good catch however the implementation does not distinguish the event that'll be fired. Running:
```ruby
batch.on(:complete, 'AnotherClass#method', 'uid' => current_user.id)
```
will run `method` on both `complete` and `success`.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
-require "codeclimate-test-reporter"
-CodeClimate::TestReporter.start
+require 'simplecov'
+SimpleCov.start
```
Comment (managr) [inline spec/spec_helper.rb]: Can you also revert this? It's not related to callback change.

File: lib/sidekiq/batch/callback.rb [hunk 2]
```diff
@@ -6,9 +6,10 @@ class Worker
 
         def perform(clazz, event, opts, bid)
           return unless %w(success complete).include?(event)
-          instance = clazz.constantize.send(:new) rescue nil
-          return unless instance
-          instance.send("on_#{event}", Sidekiq::Batch::Status.new(bid), opts) rescue nil
+          clazz, method = clazz.split("#") if (clazz.class == String && clazz.include?("#"))
+          method = "on_#{event}" if method.nil?
+          instance = clazz.constantize rescue nil
```
Comment (managr) [inline lib/sidekiq/batch/callback.rb]: Name `instance` is now misleading since it's not instance.

File: lib/sidekiq/batch/callback.rb [hunk 3]
```diff
@@ -6,9 +6,10 @@ class Worker
 
         def perform(clazz, event, opts, bid)
           return unless %w(success complete).include?(event)
```
Comment (managr) [inline lib/sidekiq/batch/callback.rb]: It doesn't really care now about success or complete, so it won't work for
```ruby
batch.on(:complete, 'AnotherClass#method', 'uid' => current_user.id)
```
Will run both for `complete` and `success`

File: spec/spec_helper.rb (hunk 1 above)
Comment (cybey) [inline spec/spec_helper.rb]: I agree. However, when I do not add this Github travis will fail since tests refuse to run. Could you add them yourself?

Comment (managr) [inline spec/spec_helper.rb]: Sure. Will take care of it.

File: lib/sidekiq/batch/callback.rb [hunk 4]
```diff
@@ -5,10 +5,9 @@ class Worker
         include Sidekiq::Worker
 
         def perform(clazz, event, opts, bid)
-          return unless %w(success complete).include?(event)
```
Comment (managr) [inline lib/sidekiq/batch/callback.rb]: Actually I was little wrong with the previous comment. Can we have that line above just to be safe?

**Comment by managr** [comment]
@cybey that's a good work. By any chance can you add a spec that covers the scenario when callback has a method name (specified with `#`)?

**Comment by managr** [comment]
Hey @cybey can you catch up with latest master? (It has the fixes related to codeclimate/simplecov). Thanks!

**Pull request merged**